wiki/osu!wiki_contribution_guide/en.md

```diff
@@ -1,178 +1,895 @@
 
-After making your changes locally, you will need to commit this on your local copy.
+Open up Git Bash or your terminal of choice (if you have not), and type in this command in the CLI:
+
+```prolog
+git status
+```
+
+This command is used to check for Git's response about the status of the local copy in this directory.
+
+If it throws up an error (meaning, you are in the wrong directory), type in this command in the CLI:
+
+```prolog
+cd $HOME
+cd osu-wiki
+```
+
+Assuming you are following the guide and did not do any modification to Git Bash/terminal (saving the cloned osu-wiki repository in your home's root folder):
+
+- The first command tells the terminal to go back to your home directory, `~` (if you somehow end up somewhere).
+  - An alternative method is to reopen the Git Bash/terminal again.
+- The second command, using the terminal command, `cd`, will move the terminal down to ``osu-wiki`` if found, which is where your freshly created local copy is at.
+
+There should be a light-blue coloured brackets with text inside (``master``) at the end of the line when done right in Git Bash.
+That is the branch name your local copy is currently in.
+
+You can now proceed to make your changes.
+
+#### Making a new branch
+
+To check for available branches, enter the command below in the CLI:
+
+```prolog
+git branch
+```
+
+This command will show all the branches of the remote repository (namely, your forked repo, nicknamed ``origin`` by default) copied into your local copy.
+If you are starting fresh, you will be in ``(master)`` branch by default.
+
+To switch to a branch, enter the command below in the CLI:
+
+```prolog
+git checkout <branchname>
+```
+
+Once completed, your local copy will be updated exactly to the branch copy version you switched to.
+You are now in the branch specified in ``<branchname>``.
+If you forgot to save the files before switching, try and see if your text editor still keeps it or not.
+
+To make a new branch, enter the commands below in the CLI:
+
+```prolog
+git checkout master
+git checkout -b <branchname>
+git push origin <branchname>
+```
+
+- The first command returns the Git back to ``master`` branch.
+  If you are already in ``master`` branch, nothing happens.
+- The second command makes the ``<branchname>`` branch copy based on ``master`` branch.
+- The third command updates your forked remote copy (``origin``, remember?) with a new ``<branchname>`` (make sure you are connected to the Internet before pushing anything).
+
+You are now in the new branch you created and its time to make some local changes.
+Go ahead and mess with the ``osu-wiki`` directory content.
+
+#### Updating the branch content
+
+After making your changes locally (such as add/delete/rename/move/modify files/folders), you will need to commit (update) this on your local copy.
+
+This can be done by entering this to the Git's CLI:
 
-This can be done by entering this to the Command Line Interface (CLI): 
 ```prolog
 git add .
```

Comment:
`git add .` will add current directory(and subdirectory) ,not whole repo(if you are not in `osu-wiki/`)